diff --git a/src/compat.ml b/src/compat.ml
index 9a96d0e..ebc3e0e 100644
--- a/src/compat.ml
+++ b/src/compat.ml
@@ -1,2 +1,6 @@
-let concat_map = List.concat_map
-let find_map = List.find_map
+let concat_map f l = List.concat (List.map f l)
+let rec find_map f = function
+  | [] -> None
+  | x::rst -> match f x with
+    | Some x -> Some x
+    | _ -> find_map f rst
diff --git a/src/entry.ml b/src/entry.ml
index 2c86491..701d2e9 100644
--- a/src/entry.ml
+++ b/src/entry.ml
@@ -62,7 +62,7 @@ let entry_of_signature lf (s : Types.signature_item)
   in
 
   let expr_to_string = function
-    | Pexp_constant (Pconst_string (str, _, _)) -> Some str
+    | Pexp_constant (Pconst_string (str, _)) -> Some str
     | _ -> None in
 
   let struct_to_string = function
diff --git a/theories/Data/Seq.v b/theories/Data/Seq.v
index 4521dee..a7563b8 100644
--- a/theories/Data/Seq.v
+++ b/theories/Data/Seq.v
@@ -79,10 +79,6 @@ Module SeqExtraction.
     [ "Stdlib.Seq.Nil" "Stdlib.Seq.Cons" ].
   Extract Constant empty => "Stdlib.Seq.empty".
   Extract Constant ret => "Stdlib.Seq.return".
-  (* since OCaml 4.11 *)
-  Extract Constant cons => "Stdlib.Seq.cons".
-  (* since OCaml 4.11 *)
-  Extract Constant append => "Stdlib.Seq.append".
   Extract Constant map => "Stdlib.Seq.map".
   Extract Constant filter => "Stdlib.Seq.filter".
   Extract Constant filter_map => "Stdlib.Seq.filter_map".
