diff --git a/src/entry.ml b/src/entry.ml
index 94afa51..3b4e4f2 100644
--- a/src/entry.ml
+++ b/src/entry.ml
@@ -85,7 +85,7 @@ let polymorphic_params (decl : type_declaration) : string list =
 
   let existing_params params =
     let polymorphic_param (t : type_expr) : string option =
-      match Types.get_desc t with
+      match t.desc with
       | Tvar (Some "_") | Tvar None -> None
       | Tvar (Some x) -> Some x
       | _ ->
@@ -97,7 +97,6 @@ let polymorphic_params (decl : type_declaration) : string list =
 
   Compat.fold_left_map
     (fun params t ->
-      let t = Types.Transient_expr.coerce t in
       match t.desc with
       | Tvar (Some "_") | Tvar None ->
           let x, params = pick_param params in
diff --git a/src/repr.ml b/src/repr.ml
index 112565f..57a1107 100644
--- a/src/repr.ml
+++ b/src/repr.ml
@@ -130,7 +130,7 @@ let pick_params =
 let named_poly_vars (t : Types.type_expr) : string list =
   let minimize = List.sort_uniq String.compare in
   let rec named_poly_vars (t : Types.type_expr) =
-    match Types.get_desc t with
+    match t.desc with
     | Tvar (Some "_") | Tvar None -> []
     | Tvar (Some x) -> [ x ]
     | Tarrow (_, t1, t2, _) ->
@@ -147,7 +147,7 @@ let mono_type_repr_of_type_expr_with_params params t :
     params_pool * mono_type_repr =
   let initial_pos = 1 in
   let rec aux pos params (t : Types.type_expr) =
-    match Types.get_desc t with
+    match t.desc with
     | Tvar (Some "_") | Tvar None ->
         let p, params = pick_param params in
         (params, TParam (CName p, []))
@@ -289,7 +289,7 @@ let all_poly_vars params t : string list =
   let minimize = List.sort_uniq String.compare in
 
   let rec poly_vars params (t : Types.type_expr) : params_pool * string list =
-    match Types.get_desc t with
+    match t.desc with
     | Tvar (Some "_") | Tvar None ->
         let x, params = pick_param params in
         (params, [ x ])
