diff --git a/src/compat.ml b/src/compat.ml
index 9a96d0e..ebc3e0e 100644
--- a/src/compat.ml
+++ b/src/compat.ml
@@ -1,2 +1,6 @@
-let concat_map = List.concat_map
-let find_map = List.find_map
+let concat_map f l = List.concat (List.map f l)
+let rec find_map f = function
+  | [] -> None
+  | x::rst -> match f x with
+    | Some x -> Some x
+    | _ -> find_map f rst
diff --git a/src/entry.ml b/src/entry.ml
index 6e66c07..1c19ba7 100644
--- a/src/entry.ml
+++ b/src/entry.ml
@@ -48,7 +48,7 @@ let entry_of_signature ?(transparent_types = false) (s : Types.signature_item)
     List.exists is_impure in
 
   let expr_to_string = function
-    | Pexp_constant (Pconst_string (str, _, _)) -> Some str
+    | Pexp_constant (Pconst_string (str, _)) -> Some str
     | _ -> None in
 
   let struct_to_string = function
